drafts/str4d-overwinter-activation/draft1.rst

```diff
@@ -0,0 +1,232 @@
+
+Hard fork
+  The creation of a new branch by a change in the consensus rules of the network. Nodes that do not recognize
+  the new rules will continue to follow the old branch.
+
+Network Upgrade
+  An intentional hard fork undertaken by the community in order to improve the network.
+
+
+Abstract
+========
+
+This proposal defines a mechanism for coordinating upgrades of the Zcash network, in order to remove ambiguity
+about when network upgrades will activate, provide defined periods in which users should upgrade their local
+software, and minimize the risks to both the upgrading network and those users opting out of the changes.
+
+
+Motivation
+==========
+
+Zcash is a *consensual currency*: nobody is ever going to force someone to use a specific software
+implementation or a specific branch of Zcash. [#consensual-currency]_ As such, different sub-communities will
+always have the freedom to choose different variants or branches which offer different design trade-offs.
+
+The current Zcash software includes an *auto-senescence* feature, causing nodes running a particular version
+to automatically shut down 16 weeks after that version was released. This was implemented for several reasons
+[#release-lifecycle]_:
+
+- It gives the same systemic advantage of removing old software as auto-upgrade behavior.
+
+- It requires users to individually choose one of the following options:
+
+  - Upgrade to a more recent software release from the main network.
+
+  - Upgrade to an alternative release.
+
+  - Ignore the "deprecation" and keep running the older software.
+
+Developers can rely on this cadence for coordinating network upgrades. Once the last pre-upgrade software
+version has been deprecated, they can reasonably assume that all node operators on the network either support
+the upgraded rules, or have explicitly chosen not to follow them.
+
+However, this behaviour is not sufficient for performing network upgrades. A globally-understood on-chain
+activation mechanism is necessary so that nodes can unambiguously know at what point the changes from an
+upgrade come into effect (and can enforce consensus rule changes, for example).
+
+
+Specification
+=============
+
+The following constants are defined for every network upgrade:
+
+BRANCH_ID
+  A globally-unique non-zero 32-bit identifier.
+
+  Implementations MAY use a value of zero in branch ID fields to indicate the absence of any upgrade (i.e.
+  that the Sprout consensus rules apply).
+
+ACTIVATION_HEIGHT
+  The block height at which the network upgrade rules will come into effect, and be enforced as part of the
+  blockchain consensus.
+
+  For removal of ambiguity, the block at height ``ACTIVATION_HEIGHT - 1`` is subject to the pre-upgrade
+  consensus rules, and would be the last common block in the event of a persistent pre-upgrade branch.
+
+  It MUST be greater than the value of ``DEPRECATION_HEIGHT`` in the last software version that will not
```

Comment:
``DEPRECATION_HEIGHT`` of what exactly? This term isn't defined in the ZIP.
Also "the last software version that will not contain support for this network upgrade" seems complicated and not necessarily well-defined. e.g. if I change one line in the network upgrade isn't a different upgrade, or if multiple people are making clients.

Reply:
Added a definition to the Motivation section.

Reply:
> e.g. if I change one line in the code related to the network upgrade is it a different upgrade?

Yes, because the consensus rules resulting from that upgrade (which is what the network upgrade is defined as changing) would be different.

> or if multiple people are making clients.

That's why we need to have ZIPs and test cases 😄

Reply:
So `DEPRECATION_HEIGHT` is one global constant of a software version? Whereas `ACTIVATION_HEIGHT` is specific to a network upgrade object, so there are many of them?
I would emphasize this.
Also, it seems you're saying the next version activates only after the previous version was deprecated, and there's a time period where no version is active.

Reply:
Ohh I see what you're saying. I wonder if "than any version that doesn't contain this upgrade"  is better than "the last version that doesn't contain this upgrade"

Reply:
and perhaps add an explanation in parenthesis "(This ensures only version containing the upgrade are not deprecated during `ACTIVATION_HEIGHT`) "